addons/web/static/src/views/fields/radio/radio_field.xml

```diff
@@ -2,16 +2,27 @@
 <templates xml:space="preserve">
 
     <t t-name="web.RadioField">
+        <t t-if="props.button">
```

Comment:
Maybe extend RadioField for your use case ? With your option, the radio field don t look like a radio selection anymore